packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart

```diff
@@ -0,0 +1,204 @@
+
+void main() {
+  Directory tempDir;
+  Directory projectDir;
+
+  setUpAll(() async {
+    Cache.disableLocking();
+    await _ensureFlutterToolsSnapshot();
+  });
+
+  setUp(() {
+    tempDir = globals.fs.systemTempDirectory
+        .createTempSync('flutter_tools_generated_plugin_registrant_test.');
+    projectDir = tempDir.childDirectory('flutter_project');
+  });
+
+  tearDown(() {
+    tryToDelete(tempDir);
+  });
+
+  testUsingContext('generated plugin registrant passes analysis', () async {
+    await _createProject(projectDir, <String>[]);
+    // We need to add a dependency with web support to trigger
+    // the generated_plugin_registrant generation.
+    await _addDependency(projectDir, 'shared_preferences', '^0.5.12+4');
+    await _analyzeProject(projectDir);
+
+    expect(
+        projectDir.childFile('lib/generated_plugin_registrant.dart'), exists);
+  }, overrides: <Type, Generator>{
+    Pub: () => Pub(
+          fileSystem: globals.fs,
+          logger: globals.logger,
+          processManager: globals.processManager,
+          usage: globals.flutterUsage,
+          botDetector: globals.botDetector,
+          platform: globals.platform,
+        ),
+  });
+
+  testUsingContext(
+      'generated plugin registrant ignores lines longer than 80 chars',
+      () async {
+    await _createProject(projectDir, <String>[]);
+    await _addAnalysisOptions(
+        projectDir, <String>['lines_longer_than_80_chars']);
+    await _createProject(tempDir.childDirectory('test_plugin'), <String>[
+      '--template=plugin',
+      '--platforms=web',
+      '--project-name',
+      'test_web_plugin_with_a_purposefully_extremely_long_package_name',
+    ]);
+    // The line for the test web plugin (`  TestWebPluginWithAPurposefullyExtremelyLongPackageNameWeb.registerWith(registrar);`)
+    // exceeds 80 chars.
+    // With the above lint rule added, we want to ensure that the `generated_plugin_registrant.dart`
+    // file does not fail analysis (this is a regression test - an ignore was
+    // added to cover this case).
+    await _addDependency(
+        projectDir,
+        'test_web_plugin_with_a_purposefully_extremely_long_package_name',
+        '\n        path: ../test_plugin');
+    await _analyzeProject(projectDir);
+
+    expect(
+        projectDir.childFile('lib/generated_plugin_registrant.dart'), exists);
+  }, overrides: <Type, Generator>{
+    Pub: () => Pub(
+          fileSystem: globals.fs,
+          logger: globals.logger,
+          processManager: globals.processManager,
+          usage: globals.flutterUsage,
+          botDetector: globals.botDetector,
+          platform: globals.platform,
+        ),
+  });
+}
+
+Future<void> _ensureFlutterToolsSnapshot() async {
+  final String flutterToolsPath = globals.fs.path.absolute(globals.fs.path.join(
+    'bin',
+    'flutter_tools.dart',
+  ));
+  final String flutterToolsSnapshotPath =
+      globals.fs.path.absolute(globals.fs.path.join(
+    '..',
```

Comment:
formatting is confusing here. Looks like the '..' is less nested than the previous line but it's actually more nested?

If you're using `dartfmt` you can probably get a better effect by putting a comma between the two `))`s at the end of the statement (line 109 below).